CorsixTH/Lua/entities/humanoids/vip.lua

```diff
@@ -126,6 +180,55 @@ function Vip:goHome()
   self:despawn()
 end
 
+function Vip:evaluateRoom()
+  local room_extinguisher = 0
+  local room_plant = 0
+  local room_bin = 0
+  -- Another room visited.
+  self.num_visited_rooms = self.num_visited_rooms + 1
+  local room = self.next_room
+  -- if the player is about to kill a live patient for research, punish hard
+  if room.room_info.id == "research" then
+    if room:getPatient() then
+      self.vip_rating = self.vip_rating + 6
+    end
+  end
+  -- evaluate the room we're currently looking at
+  for object, _ in pairs(room.objects) do
+    if object.object_type.id == "extinguisher" and room_extinguisher == 0 then
+      self.room_eval = self.room_eval + 1
+      -- only count this object type once
+      room_extinguisher = 1
+    elseif object.object_type.id == "plant" then
+      if object.days_left >= 10 then
+        room_plant = room_plant + 1
+      elseif object.days_left <= 3 then
+        room_plant = room_plant - 1
+      end
+    elseif object.object_type.id == "bin" and room_bin == 0 then
+      self.room_eval = self.room_eval + 1
+      -- only count this object type once
+      room_bin = 1
+    end
+
+    if object.strength then
```

Comment:
Same with the object strength - maybe object:isAlmostBroken() or something?

Reply:
This test seems to look at if the machine is getting old rather than technically broken. In other words it's lost 50% of its strength from wear and not been replaced.  Though I didn't write this bit of code originally. 

If we were looking at its health then we could call Machine:getRemainingUses()